File: backend/mr-db-sql/src/main/java/io/featurehub/db/publish/DummyPublisher.java
package io.featurehub.db.publish;

import io.featurehub.dacha.model.PublishAction;
import io.featurehub.db.model.DbApplicationFeature;
import io.featurehub.db.model.DbEnvironment;
import io.featurehub.db.model.DbFeatureValue;
import io.featurehub.db.model.DbRolloutStrategy;
import io.featurehub.db.model.DbServiceAccount;

import java.util.UUID;

/**
 *
 */
public class DummyPublisher implements PublishManager, CacheSource {
  @Override
  public void registerCache(String cacheName, CacheBroadcast cacheBroadcast) {

  }

  @Override
  public void publishToCache(String cacheName) {

  }

  @Override
  public void publishFeatureChange(DbFeatureValue strategy) {

  }

  @Override
  public void deleteFeatureChange(DbApplicationFeature feature, UUID environmentId) {

  }

  @Override
  public void updateServiceAccount(DbServiceAccount serviceAccount, PublishAction publishAction) {

  }

  @Override
  public void deleteServiceAccount(UUID id) {

  }


  @Override
  public void updateEnvironment(DbEnvironment environment, PublishAction publishAction) {

  }

  @Override
  public void deleteEnvironment(UUID id) {

  }

  @Override
  public void publishFeatureChange(DbApplicationFeature appFeature, PublishAction update) {

  }

  @Override
  public void publishFeatureChange(DbApplicationFeature appFeature, PublishAction update, String featureKey) {

  }

  @Override
  public void publishRolloutStrategyChange(DbRolloutStrategy rs) {

  }


}

File: backend/mr-eventing/src/main/kotlin/io/featurehub/db/publish/DummyPublisher.kt
package io.featurehub.db.publish

import io.featurehub.dacha.model.PublishAction
import io.featurehub.db.api.CacheRefresherApi
import io.featurehub.db.model.*
import io.featurehub.mr.events.common.CacheSource
import java.util.*

/**
 *
 */
class DummyPublisher : CacheSource, CacheRefresherApi {
  override fun publishObjectsAssociatedWithCache() {}
  override fun publishFeatureChange(featureValue: DbFeatureValue) {}
  override fun publishApplicationRolloutStrategyChange(update: PublishAction, rs: DbApplicationRolloutStrategy) {
  }

  override fun deleteFeatureChange(feature: DbApplicationFeature, environmentId: UUID) {}
  override fun updateServiceAccount(serviceAccount: DbServiceAccount, publishAction: PublishAction) {}
  override fun deleteServiceAccount(id: UUID) {}
  override fun updateEnvironment(environment: DbEnvironment, publishAction: PublishAction) {}
  override fun deleteEnvironment(id: UUID) {}
  override fun publishFeatureChange(appFeature: DbApplicationFeature, update: PublishAction) {}
  override fun publishFeatureChange(appFeature: DbApplicationFeature, update: PublishAction, featureKey: String) {}
  override fun publishApplicationRolloutStrategyChange(rs: DbApplicationRolloutStrategy) {}
  override fun refreshPortfolios(portfolioIds: List<UUID>) {
  }

  override fun refreshApplications(applicationIds: List<UUID>) {
  }

  override fun refreshEntireDatabase() {
  }
}
